packages/contractkit/src/explorer/block-explorer.ts

```diff
@@ -0,0 +1,122 @@
+import { Address } from '@celo/utils/lib/address'
+import abi, { ABIDefinition } from 'web3-eth-abi'
+import { Block, Transaction } from 'web3/eth/types'
+import { ContractKit } from '../kit'
+import { ContractDetails, obtainKitContractDetails } from './base'
+
+export interface CallDetails {
+  contract: string
+  function: string
+  parameters: Record<string, any>
+}
+
+export interface ParsedTx {
+  callDetails: CallDetails
+  tx: Transaction
+}
+
+export interface ParsedBlock {
+  block: Block
+  parsedTx: ParsedTx[]
+}
+
+interface ContractMapping {
+  details: ContractDetails
+  fnMapping: Map<string, ABIDefinition>
+}
+
+export async function newBlockExplorer(kit: ContractKit) {
+  return new BlockExplorer(kit, await obtainKitContractDetails(kit))
+}
+
+export class BlockExplorer {
+  private addressMapping: Map<Address, ContractMapping> = new Map()
+
+  constructor(private kit: ContractKit, readonly contractDetails: ContractDetails[]) {
+    for (const cd of contractDetails) {
+      const fnMapping: Map<string, ABIDefinition> = new Map()
+      for (const abiDef of cd.jsonInterface as ABIDefinition[]) {
+        if (abiDef.type === 'function') {
+          fnMapping.set(abiDef.signature, abiDef)
+        }
+      }
+
+      this.addressMapping.set(cd.address, {
+        details: cd,
+        fnMapping,
+      })
+    }
+  }
+
+  async fetchBlockByHash(blockHash: string): Promise<Block> {
+    // TODO fix typing: eth.getBlock support hashes and numbers
+    const block = await this.kit.web3.eth.getBlock(blockHash as any, true)
```

Comment:
you can just return `this.kit.web3.eth.getBlock(blockHash as any, true)` straight?